opm/autodiff/StandardWells_impl.hpp

```diff
@@ -1248,4 +1248,304 @@ namespace Opm
+           // the potential limits should not be difficult to add
+           const WellEcon::QuantityLimitEnum& quantity_limit = econ_production_limits.quantityLimit();
+           if (quantity_limit == WellEcon::POTN) {
+               const std::string msg = std::string("POTN limit for well ") + well_name + std::string(" is not supported for the moment. \n")
+                                     + std::string("All the limits will be evaluated based on RATE. ");
+               OpmLog::warning(msg);
+           }
+
+           const WellMapType& well_map = well_state.wellMap();
+           typename WellMapType::const_iterator i_well = well_map.find(well_name);
+           assert(i_well != well_map.end()); // should always be found?
+           const int well_number = (i_well->second)[0];
+
+           if (econ_production_limits.onAnyRateLimit()) {
+               rate_limit_violated = checkRateEconLimits(econ_production_limits, well_state, well_number);
+           }
+
+           if (rate_limit_violated) {
+               if (econ_production_limits.endRun()) {
+                   const std::string warning_message = std::string("ending run after well closed due to economic limits is not supported yet \n")
+                                                     + std::string("the program will keep running after ") + well_name + std::string(" is closed");
+                   OpmLog::warning(warning_message);
+               }
+
+               if (econ_production_limits.validFollowonWell()) {
+                   OpmLog::warning("opening following on well after well closed is not supported yet");
+               }
+
+               if (well_ecl->getAutomaticShutIn()) {
+                   list_econ_limited.addShutWell(well_name);
+                   const std::string msg = std::string("well ") + well_name + std::string(" will be shut in due to economic limit");
+                   OpmLog::info(msg);
+               } else {
+                   list_econ_limited.addStoppedWell(well_name);
+                   const std::string msg = std::string("well ") + well_name + std::string(" will be stopped due to economic limit");
+                   OpmLog::info(msg);
+               }
+               // the well is closed, not need to check other limits
+               continue;
+           }
+
+           // checking for ratio related limits, mostly all kinds of ratio.
+           int worst_offending_connection = -1e4;
+           bool ratio_limits_violated = false;
+           bool last_connection = false;
+
+           if (econ_production_limits.onAnyRatioLimit()) {
+               ratio_limits_violated = checkRatioEconLimits(econ_production_limits, well_state, i_well,
+                                                            worst_offending_connection, last_connection);
+           }
+
+           if (ratio_limits_violated) {
+               const int perf_start = (i_well->second)[1];
+               const int perf_number = (i_well->second)[2];
```

Comment:
This variable is only used in the assertation below, so it will be unused in release mode and generate a warning. Avoid that warning, for example by using the expression `(i_well->second)[2]` directly in the assert().